### PR TITLE
Additional Time Options. TIME LAPSE

### PR DESCRIPTION
**Pull request by dratom**
Added requested time options:
*moving days forward / backward
*incrementing time 5 minutes forward / backward

Added Time Lapse toggle that increments time ahead 1min open endedly.

### CODE REVIEW

**Comment by dratom** [comment]
Added 1 hour functions

**Comment by dratom** [comment]
1 hour functions have been assimilated.
Resistance is futile.

**Comment by dratom** [comment]
It is what it is. At least keep the day forward / backward code. I'm pretty proud of myself for writing it out instead of pulling from elsewhere as im sure there are oodles of calendar scripts.

**Comment by gtav-ent** [comment]
Remind me, what is Time Lapse meant to do?

**Comment by dratom** [comment]
Not much more then look pretty. It speeds the day by one minute at a time so that the the speedup nature of what you see looks seamless. Naming it "Time Lapse" is pretty misleading but to be honest I put very little thought into renaming this before release. When I initially write things out I give them pretty obnoxious names that are not very family friendly even when censored.

**Comment by gtav-ent** [comment]
All your stuff is in now, time lapse excepted, but it was done via C&P rather than a merge. I refactored it a little, but it's mostly your work.

I did find some weirdness with the game and month boundaries; on changing the date forward, the game sometimes changes it forward again. The fix seemed to be to reapply the current time as well.

**Comment by dratom** [comment]
Thank You x100 for doing that. I honestly had closed this because I expected that my work was not up to par and was wiping my fork to go back to simply adding little fixes and making changes locally before compile.

That was an error I thought I had corrected. I found that zeroing the seconds field every time I did a time or date change was the trick. There was also a problem with the day of the week not moving forward but it turned out to be that the in game phone (the only place in game that displays the day fo the week) did not update the displayed day of the week unless you allowed an in game minute to pass. But I didn't do as much testing after I moved my stuff over to the new menu class layout.

tl:dr

Thank You for porting it over and my bad about the bugs.

**Comment by gtav-ent** [comment]
Ah don't be daft. A few people seem to think that, but really integration delays are just about finding a point in time that suits merging it in, allows time to test and doesn't introduce too much risk to whatever else is going on. Qais' menu thing took weeks to get actioned, but didn't really involve any serious work when it was time.

Off the top of my head, things I did to yours:

- put it into methods outside of the per-frame update_features() method, since it only happens once, and so got rid of the feature flags too

- fixed a logical bug involving boundaries and 24+ hour times etc

- tidied up some logic, so it does the same job, just more neatly

- added a calendar change when you go forward/back over a midnight boundary

- added some stuff to fix game weirdness as mentioned. Incidentally have you noticed that the game days don't match the date? I think sometimes you even move forward a calendar date and the day stays the same, or skips one. Weird. 

That sounds like a lot but it's not really. It's much better to start with that and make a few tweaks than it is to do it all from scratch.

**Comment by dratom** [comment]
Thank You for the detailed explanation. 
I could swear I had put calls to the calendar forward/backward functions with the 5min/1hour forward/backward loops. That was the only reason I kept the loops. also the reason I added the isPlayerCalled global to keep the date popup from displaying when date changes are called internally. I was also extra careful with the 2359 -> 0000 and 2359 <- 0000 portions as the game will insta-crash if you so much as think about 2400 too hard. At least it does for me.
My apologies on the the bugs. I played around in game for at least an hour before I finally pushed those changes. I understand that it wasn't much for you to have to debug all that, but it bothers me that after taking my time I missed. 
Perhaps by now it's apparent that C++ and I are new friends.  However I eat sleep and breathe Verilog(SoC RnD) and was under the impression that Logic and I were old lovers. But now I find out that girl done keyed my car and put thumb tacks in my shoes? I'm gonna have to have a talk with her. My apologies good sir but it is past my hour to retire and I do get a byte loopy after that.
